### ISSUE TITLE
_INT64 functions

### ISSUE THREAD

**Issue by vbmithr** [DTC-protocol/DTC#9]
So, DTC supports mostly messages taking quantities as either double (the default), float, or int32.

I'd like to use integer price format (because I don't want to have do deal with the floating point errors), but the int32 type is too small for me (for example, some bitcoin exchanges accepts volumes in satoshis, meaning that 100 bitcoins in satoshis overflows an int32).

Would you mind considering adding INT64 functions too ?

**Comment by SierraChart** [comment]
Fork the project. Define the new market data message constants, data structures, and member functions. When done, make a pull request to us.

Make sure you are working with the latest files. We have just updated them.

**Comment by SierraChart** [comment]
Do you still want to support the 64-bit integer messages? 

The concern about them is that it is more work for a client to implement DTC with so many different message types for market data.   We do not want to scare people away from this protocol.  

Sierra Chart, the client that we work on, does not support this type internally so there is no benefit regarding SC. And 64 byte floats which are already supported should give sufficient precision even though are not 100% precise.

If they are supported, it is necessary to implement all the associated functions within the CPP file for them. However, Sierra Chart will not support these particular messages  other than possibly for market data. But not for trading.

**Comment by SierraChart** [comment]
It is also necessary to prepare documentation for them. So this is another thing that has to be done.

**Comment by vbmithr** [comment]
Ok, I don't care this much. I just find it stupid to use floating point numbers to represent exact quantities. Floating point is for real numbers, and there is not a single real number in the context of finance (I'm not talking about computation on financial data of course).

If I wanted to simplify the protocol, I would get rid of everything else than int64.
I leave the final choice to you, though I agree 4 different message types is too much.

It is also possible to define 2 message types, one with 32bit fields and the other 64 bits, with a boolean field to discriminate between float and int types.

So to sum up, I don't care, but I want to stress that I find stupid to use floating point numbers where quantities described are exact fraction of integers. Unless there are a good reason to do so (which I don't know about).

**Comment by SierraChart** [comment]
We do agree with what you say.

For the integer order messages, we will change those to 64-bit.

The problem with market data messages is that we could change the 32-bit Integers to 64-bit, but then there is the problem of size which is a major consideration when it comes to network bandwidth and high-frequency data.

So we will implement  64-bit price and quantities,  but we just need to think through this some more and implement them in the best most reasonable way.

**Comment by vbmithr** [comment]
On 31/07/2015 02:47, DTC Protocol Engineering wrote:
> So we will implement 64-bit price and quantities, but we just need to
> think through this some more and implement them in the best most
> reasonable way.

Ok, great!
